lib/rules/no-unreachable.js

```diff
@@ -151,8 +151,11 @@ module.exports = {
             /*
              * Report the current range since this statement is reachable or is
              * not consecutive to the current range.
+             * Empty statements are not considered unreachable as they do not
+             * contains executable code. See
+             * https://github.com/eslint/eslint/issues/9081
              */
-            if (!range.isEmpty) {
+            if (!range.isEmpty && range.startNode.type !== "EmptyStatement") {
```

Comment:
This change makes the rule ignoring unreachable range which starts with `EmptyStatement`. So it will overlook the following case:

```js
function f(a) {
    switch (a) {
        case 0: return 1;
        default: return 2;
    };
    g() //← `; g()` starts with `;`.
}
```

Please delete L188 `EmptyStatement: reportIfUnreachable,` instead.